sql/core/src/main/scala/org/apache/spark/sql/test/ExamplePointUDT.scala

```diff
@@ -27,6 +27,12 @@ import org.apache.spark.sql.types._
  */
 @SQLUserDefinedType(udt = classOf[ExamplePointUDT])
 private[sql] class ExamplePoint(val x: Double, val y: Double) extends Serializable {
+
+  override def hashCode(): Int = {
+    val state = Seq(x, y)
+    state.map(_.hashCode()).foldLeft(0)((a, b) => 31 * a + b)
```

Comment:
I don't really mind this, but I think this is overkill when there are 2-3 fields. This could be `31 * x.hashCode() + y.hashCode()`